### PR TITLE
[DMC] Implement Ohabi Caleria

### PR DESCRIPTION
**Pull request by PurpleCrowbar**

### CODE REVIEW

**Comment by PurpleCrowbar** [comment]
[[Ohabi Caleria]]

File: Mage.Sets/src/mage/cards/o/OhabiCaleria.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        super(card);
+    }
+
+    @Override
+    public OhabiCaleria copy() {
+        return new OhabiCaleria(this);
+    }
+}
+
+class OhabiCaleriaTriggeredAbility extends TriggeredAbilityImpl {
```
Comment (awjackson) [inline Mage.Sets/src/mage/cards/o/OhabiCaleria.java:66]: This custom ability is a duplicate of `DealsDamageToACreatureAllTriggeredAbility`.

**Review by awjackson — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/o/OhabiCaleria.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import mage.game.permanent.Permanent;
+
+import java.util.UUID;
+
+/**
+ * @author PurpleCrowbar
+ */
+public final class OhabiCaleria extends CardImpl {
+
+    private static final FilterCreaturePermanent filter = new FilterCreaturePermanent("Archers you control");
```
Comment (awjackson) [inline Mage.Sets/src/mage/cards/o/OhabiCaleria.java:30]: This needs to be `FilterControlledCreaturePermanent` if you're using the common ability class. And you can put the Subtype right in the constructor for both `FilterCreaturePermanent` and `FilterControlledCreaturePermanent`.

Comment (PurpleCrowbar) [reply Mage.Sets/src/mage/cards/o/OhabiCaleria.java]: Done. However, I needed an additional filter for the correct message to display in the context of the third ability

File: Mage.Sets/src/mage/cards/o/OhabiCaleria.java [hunk 3]
```diff
@@ -48,9 +38,9 @@ public OhabiCaleria(UUID ownerId, CardSetInfo setInfo) {
         this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new UntapAllDuringEachOtherPlayersUntapStepEffect(filter)));
 
         // Whenever an Archer you control deals damage to a creature, you may pay {2}. If you do, draw a card.
-        this.addAbility(new OhabiCaleriaTriggeredAbility(new DoIfCostPaid(
+        this.addAbility(new DealsDamageToACreatureAllTriggeredAbility(new DoIfCostPaid(
                 new DrawCardSourceControllerEffect(1), new GenericManaCost(2)
-        )));
+        ), false, filter2, SetTargetPointer.PERMANENT, false));
```
Comment (awjackson) [inline Mage.Sets/src/mage/cards/o/OhabiCaleria.java:43]: This should be `SetTargetPointer.NONE`. `DrawCardSourceControllerEffect` is not a targetted effect.

File: Mage.Sets/src/mage/cards/o/OhabiCaleria.java (hunk 2 above)
Comment (awjackson) [reply Mage.Sets/src/mage/cards/o/OhabiCaleria.java]: That's fine.

File: Mage.Sets/src/mage/cards/o/OhabiCaleria.java (hunk 3 above)
Comment (PurpleCrowbar) [reply Mage.Sets/src/mage/cards/o/OhabiCaleria.java]: Done

**Review by awjackson — APPROVED** [review]

**Pull request merged**